src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/AlleleSubsettingUtils.java

```diff
@@ -0,0 +1,233 @@
+     * @param vc                                variant context with alleles and genotype likelihoods
+     * @param defaultPloidy                     ploidy to assume in case that {@code vc} does not contain that information
+     *                                          for a sample.
+     * @param allelesToKeep                      alleles to subset
+     * @param method                            whether to assign hard genotypes or leave as no-call
+     * @return                                  GenotypesContext with new PLs
+     */
+    public static GenotypesContext subsetAlleles(final VariantContext vc, final int defaultPloidy,
+                                                 final List<Allele> allelesToKeep, final GenotypeAssignmentMethod method) {
+        Utils.nonNull(vc, "vc is null");
+        Utils.nonNull(allelesToKeep, "allelesToKeep is null");
+        Utils.nonEmpty(allelesToKeep, "must keep at least one allele");
+        Utils.validateArg(allelesToKeep.get(0).isReference(), "First allele must be the reference allele");
+        if (allelesToKeep.size() == 1) {
+            return GATKVariantContextUtils.subsetToRefOnly(vc, defaultPloidy);
+        }
+        final GenotypesContext newGTs = GenotypesContext.create();
+
+        // create the new genotypes
+        for ( final Genotype g : vc.getGenotypes() ) {
+            final int ploidy = g.getPloidy() > 0 ? g.getPloidy() : defaultPloidy;
+            if ( !g.hasLikelihoods() ) {
+                newGTs.add(GenotypeBuilder.create(g.getSampleName(), GATKVariantContextUtils.noCallAlleles(ploidy)));
+                continue;
+            }
+
+            final double[] originalLikelihoods = g.getLikelihoods().getAsVector();
+
+            //TODO: the subsetting can be recycled from sample to sample if the ploidy stays the same
+            final double[] newLikelihoods;
+            if (allelesToKeep.size() == vc.getNAlleles()) {
+                newLikelihoods = originalLikelihoods;  //no subsetting
+            } else {
+                final int[] oldPLIndicesOfNewGenotypes = subsettedGenotypeLikelihoodIndices(ploidy, vc.getAlleles(), allelesToKeep);
+                newLikelihoods = Arrays.stream(oldPLIndicesOfNewGenotypes).mapToDouble(idx -> originalLikelihoods[idx]).toArray();
+            }
+
+            if (!GATKVariantContextUtils.isInformative(newLikelihoods) ) {
+                newGTs.add(GenotypeBuilder.create(g.getSampleName(), GATKVariantContextUtils.noCallAlleles(ploidy)));
+                continue;
+            }
+
+            final GenotypeBuilder gb = new GenotypeBuilder(g);
+            gb.PL(newLikelihoods);
+            GATKVariantContextUtils.makeGenotypeCall(ploidy, gb, method, newLikelihoods, allelesToKeep);
+            newGTs.add(gb.make());
+        }
+        return newGTs;
+    }
+
+
+    //TODO: is this doing the same thing as the method immediately above????  Quite likely, yes!!!
```

Comment:
Can you address this TODO?

The body of this method above could simply be:

```
    return subsetAlleles(vc.getGenotypes(), vc.getAlleles(), allelesToUse, assignGenotypes);
```

It seems that one difference is that the method above does not take care of the AD annotation... but that seems rather a bug anyway. Perhaps there is more differences but its seems to me that what we should do here is to do the Union of all those changes across both method.